### DISCUSSION TITLE
Color code boxes and text inside with conditional

### DISCUSSION THREAD

**Discussion by blinkerfluessigkeit** [typst/typst#3124] [Q&A] [answered]
Let's say I want to put code inside a box. If the language is lang1, the box should be colored in with #eff0f3 and the code inside the box should be of color #ff0000. If the language is lang2, the box should be colored in with #eff0b6 and the code inside the box should be of color #00ff00. Then finally, if the code is neither lang1 nor lang2, the box should'nt be filled (or filled with #ffffff). How could this be done?

So far, I've been using this code, but the rect shares the width, inset and radius, so I feel like there should be a way to reuse these definitions for both (or all) cases. Also, I didn't find a way to cover the third case, where lang is neither lang1 nor lang2.
I've been trying to get this working for a couple of hours now and gotten nowhere, sadly.

```typ
    #show raw.where(
        lang: "lang1",
    ): it => rect(
        width: 100%,
        inset: 1.5em,
        radius: 0.5em,
        fill: rgb("#eff0f3"),
    )[
        #set text(fill: rgb("#ff0000"))
        #it
    ]
    #show raw.where(
        lang: "lang2"
    ): it => rect(
        width: 100%,
        inset: 1.5em,
        radius: 0.5em,
        fill: rgb("#eff0b6"),
    )[
        #set text(fill: rgb("#00ff00"))
        #it
    ]
```

**Comment by xkevio** [comment, marked as answer]
One approach would be to define a function that predefines the `rect` you want with the parameters you want to be changeable, like the `fill`.

```typ
#let code-box(fill: white, it) = rect(width: 100%, inset: 1.5em, radius: 0.5em, fill: fill, it)

#show raw.where(lang: none): it => {
  code-box(it)
}

#show raw.where(lang: "l1"): it => {
  set text(fill: rgb("#ff0000"))
  code-box(fill: rgb("#eff0f3"), it)
}

#show raw.where(lang: "l2"): it => {
  set text(fill: rgb("#00ff00"))
  code-box(fill: rgb("#eff0b6"), it)
}
```

**Reply by blinkerfluessigkeit** [reply]
Very elegant solution, thank you very much!
And also happy new year :)
